addons/web_editor/static/lib/odoo-editor/src/commands/commands.js

```diff
@@ -419,13 +410,6 @@ export const editorCommands = {
             }
         }
     },
-    removeFormat: editor => {
-        editor.document.execCommand('removeFormat');
-        for (const node of getTraversedNodes(editor.editable)) {
-            // The only possible background image on text is the gradient.
-            closestElement(node).style.backgroundImage = '';
-        }
-    },
```

Comment:
I think this code should stay, for when we explicitely remove the format of a text we want to clear its gradient too. Your fix should only impact the removal of formats when switching tags. I guess it's the `stepFunction` of the test that you need to change instead.